liblumen_codegen/lib/lumen/compiler/Dialect/EIR/Conversion/EIRToLLVM/ConvertEIRToLLVM.cpp

```diff
@@ -263,90 +245,51 @@ static Value do_unbox_list(OpBuilder &builder, edsc::ScopedContext &context,
-                           TargetInfo &targetInfo_,
-                           mlir::PatternBenefit benefit = 1)
-      : mlir::OpConversionPattern<Op>::OpConversionPattern(context, benefit),
-        dialect(converter_.getDialect()),
-        typeConverter(converter_),
-        targetInfo(targetInfo_) {}
-
- protected:
-  LLVMTypeConverter &typeConverter;
-  TargetInfo &targetInfo;
-  LLVMDialect *dialect;
-
-  LLVMType getUsizeType() const { return targetInfo.getUsizeType(); }
-  LLVMType getI1Type() const { return targetInfo.getI1Type(); }
-  LLVMType getI32Type() const { return LLVMType::getIntNTy(dialect, 32); }
-
-  LLVMType getTupleType(ArrayRef<LLVMType> elementTypes) const {
-    return targetInfo.makeTupleType(dialect, elementTypes);
-  }
-
-  Type getIntegerType(OpBuilder &builder) const {
-    return builder.getIntegerType(targetInfo.pointerSizeInBits);
-  }
-
-  Attribute getIntegerAttr(OpBuilder &builder, int64_t i) const {
-    return builder.getIntegerAttr(getIntegerType(builder), i);
-  }
-
-  Attribute getIntegerAttr(OpBuilder &builder, APInt &i) const {
-    return builder.getIntegerAttr(getIntegerType(builder), i.getLimitedValue());
-  }
-
-  Attribute getI32Attr(OpBuilder &builder, int64_t i) const {
-    return builder.getIntegerAttr(builder.getIntegerType(32), i);
-  }
-
-  FlatSymbolRefAttr getOrInsertFunction(
-      PatternRewriter &builder, ModuleOp mod, StringRef name, LLVMType retTy,
-      ArrayRef<LLVMType> argTypes = {}) const {
-    return createOrInsertFunction(builder, mod, dialect, targetInfo, name,
-                                  retTy, argTypes);
-  }
-
-  Value processAlloc(PatternRewriter &builder, edsc::ScopedContext &context,
-                     ModuleOp parentModule, Location loc, LLVMType ty,
-                     Value allocBytes) const {
-    auto ptrTy = ty.getPointerTo();
-    auto usizeTy = getUsizeType();
-    auto callee = getOrInsertFunction(
-        builder, parentModule, "__lumen_builtin_malloc", ptrTy, {usizeTy});
-    auto call = builder.create<mlir::CallOp>(loc, callee, ArrayRef<Type>{ptrTy},
-                                             ArrayRef<Value>{allocBytes});
-    return call.getResult(0);
-  }
+static Value do_mask_immediate(OpBuilder &builder, edsc::ScopedContext &context,
+                               TargetInfo &targetInfo, Value val) {
+  auto maskInfo = targetInfo.immediateMask();
 
-  Value make_list(OpBuilder &builder, edsc::ScopedContext &context,
-                  Value cons) const {
-    return do_make_list(builder, context, typeConverter, targetInfo, cons);
+  auto termTy = targetInfo.getUsizeType();
+  auto intTy = builder.getIntegerType(targetInfo.pointerSizeInBits);
+  auto maskAttr = builder.getIntegerAttr(intTy, maskInfo.mask);
+  auto tagAttr = builder.getIntegerAttr(intTy, 4ull << 47);
+  Value tagConst = llvm_constant(termTy, tagAttr);
+  Value tagged = llvm_or(val, tagConst);
+  return tagged;
+  /*
```

Comment:
Delete or needed for other arches?

Reply:
Needed, there is some hardcoded stuff in there I need to swap out. This module is in a lot of flux, so if there is commented code anywhere, it's because I'm pretty sure I want it, but didn't need it at the moment, or it was subtly broken and I haven't gotten around to fixing it just yet.